src/search/searcher.h

```diff
@@ -188,10 +188,12 @@ class Searcher : public std::enable_shared_from_this<Searcher> {
 
         movegen::ValidMoves captures;
         core::getAllMoves<movegen::MoveCapture>(board, captures);
+        const auto prevMove = m_stackItr == m_stack.begin() ? std::nullopt : std::optional<movegen::Move>((m_stackItr - 1)->move);
```

Comment:
`std::make_optional` can auto deduce type
```suggestion
        const auto prevMove = m_stackItr == m_stack.begin() ? std::nullopt : std::make_optional((m_stackItr - 1)->move);
```